File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/java/robots/SecondScreenRobot.java
package pavelnazimok.uitestingfeatures.java.robots;

import pavelnazimok.uitestingfeatures.R;

import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;
import static pavelnazimok.uitestingfeatures.java.utils.UiAutomatorExtensions.findById;
import static pavelnazimok.uitestingfeatures.java.utils.UiAutomatorExtensions.waitUntilNotDisplayedById;

@SuppressWarnings("UnusedReturnValue")
public class SecondScreenRobot {

    public SecondScreenRobot clickAlertDialog() {
        clickOn(R.id.button_alert_dialog);
        return this;
    }

    public SecondScreenRobot clickOkInAlertDialog() {
        findById(android.R.id.button1, 10000).click();
        return this;
    }

    public SecondScreenRobot clickCancelInAlertDialog() {
        findById(android.R.id.button2, 10000).click();
        return this;
    }

    public SecondScreenRobot assertAlertDialogIsNotDisplayed() {
        waitUntilNotDisplayedById("alertTitle", 5000);
        return this;
    }
}

File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/kotlin/robots/SecondScreenRobot.kt
package pavelnazimok.uitestingfeatures.kotlin.robots

import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
import pavelnazimok.uitestingfeatures.R
import pavelnazimok.uitestingfeatures.kotlin.utils.findById
import pavelnazimok.uitestingfeatures.kotlin.utils.waitForView
import pavelnazimok.uitestingfeatures.kotlin.utils.waitUntilNotDisplayedById

fun onSecondScreen(func: SecondScreenRobot.() -> Unit): SecondScreenRobot {
    return SecondScreenRobot().apply {
        assertSecondScreenIsDisplayed()
        func()
    }
}

class SecondScreenRobot {

    fun clickAlertDialog() {
        clickOn(R.id.button_alert_dialog)
    }

    fun clickOkInAlertDialog() {
        findById(android.R.id.button1).click()
    }

    fun clickCancelInAlertDialog() {
        findById(android.R.id.button2).click()
    }

    fun assertSecondScreenIsDisplayed() {
        waitForView(R.id.screen_second)
    }

    fun assertAlertDialogIsNotDisplayed() {
        waitUntilNotDisplayedById("alertTitle")
    }
}
